### ISSUE TITLE
java_proto_library: difference between bazel7 and bazel8

### ISSUE THREAD

**Issue by tpasternak** [bazelbuild/bazel#24543]
### Description of the bug:

for java_proto_library, JavaInfo providers contents are slightly different. In bazel 7 it didn't contain any java outputs.

```
USE_BAZEL_VERSION=7.4.1 bazelisk cquery --output=starlark "--starlark:expr=providers(target)['@@_builtins//:common/java/java_info.bzl%JavaInfo'].java_outputs" //:lib

[]
```

```
USE_BAZEL_VERSION=8.0.0rc6 bazelisk cquery --output=starlark "--starlark:expr=providers(target)['@@rules_java+//java/private:java_info.bzl%JavaInfo'].java_outputs" //:lib

[<unknown object com.google.devtools.build.lib.rules.java.JavaRuleOutputJarsProvider$JavaOutput>]

```

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

File: BUILD.bazel
```py
   1 load("@protobuf//bazel:java_proto_library.bzl", "java_proto_library")
   2 
   3 java_proto_library(
   4     name = "lib",
   5     deps = [":proto_lib"],
   6 )
   7 
   8 proto_library(
   9     name = "proto_lib",
  10     srcs = ["lib.proto"],
  11 )
```

File: MODULE.bazel
```py
   1 bazel_dep(
   2     name = "protobuf",
   3     version = "29.0",
   4 )
   5 
```

File: lib.proto
```proto
   1 message Lib {
   2
   3 }   
```

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

8.0.0rc6

**Comment by hvadehra** [comment]
This regression was introduced with the proto rules moving out independently of the java rules. I'll fix in `@rules_java`

**Comment by hvadehra** [comment]
@tpasternak Could you please let us know if this issue is fixed for you with rules_java update / Bazel 8.0.0rc7 ?
